components/style/data.rs

```diff
@@ -352,10 +355,12 @@ impl ElementData {
                 shared_context,
```

Comment:
Let's invert this condition to look like:

```rust
if !element.has_snapshot() || element.handled_snapshot() {
    return InvalidationResult::empty();
}

let invalidator = ...;
let result = invalidator.invalidate();
unsafe { element.set_handled_snapshot() }
debug_assert!(element.handled_snapshot());
result
```

That way we avoid the mutability.